From: Rukkhadevata123 <rukkhadevata@foxmail.com>
Date: Fri, 7 Nov 2025 15:53:53 +0800
Subject: Allow overriding version via HMCL_VERSION env

---
 HMCL/build.gradle.kts | 32 +++++++++++++++++++-------------
 1 file changed, 19 insertions(+), 13 deletions(-)

diff --git a/HMCL/build.gradle.kts b/HMCL/build.gradle.kts
index 5c8d476..8ac244d 100644
--- a/HMCL/build.gradle.kts
+++ b/HMCL/build.gradle.kts
@@ -32,21 +32,27 @@ val curseForgeApiKey = System.getenv("CURSEFORGE_API_KEY") ?: ""
 
 val launcherExe = System.getenv("HMCL_LAUNCHER_EXE") ?: ""
 
-val buildNumber = System.getenv("BUILD_NUMBER")?.toInt()
-if (buildNumber != null) {
-    version = if (JenkinsUtils.IS_ON_CI && versionType == "dev") {
-        "$versionRoot.0.$buildNumber"
-    } else {
-        "$versionRoot.$buildNumber"
-    }
+val envReleaseVersion: String? = System.getenv("HMCL_VERSION")
+
+if (!envReleaseVersion.isNullOrBlank()) {
+    version = envReleaseVersion
 } else {
-    val shortCommit = System.getenv("GITHUB_SHA")?.lowercase()?.substring(0, 7)
-    version = if (shortCommit.isNullOrBlank()) {
-        "$versionRoot.SNAPSHOT"
-    } else if (isOfficial) {
-        "$versionRoot.dev-$shortCommit"
+    val buildNumber = System.getenv("BUILD_NUMBER")?.toInt()
+    if (buildNumber != null) {
+        version = if (JenkinsUtils.IS_ON_CI && versionType == "dev") {
+            "$versionRoot.0.$buildNumber"
+        } else {
+            "$versionRoot.$buildNumber"
+        }
     } else {
-        "$versionRoot.unofficial-$shortCommit"
+        val shortCommit = System.getenv("GITHUB_SHA")?.lowercase()?.substring(0, 7)
+        version = if (shortCommit.isNullOrBlank()) {
+            "$versionRoot.SNAPSHOT"
+        } else if (isOfficial) {
+            "$versionRoot.dev-$shortCommit"
+        } else {
+            "$versionRoot.unofficial-$shortCommit"
+        }
     }
 }
 
