From: Rukkhadevata123 <rukkhadevata@foxmail.com>
Date: Fri, 7 Nov 2025 21:49:42 +0800
Subject: remove libraries/cmark/nmake.bat

---
 libraries/cmark/nmake.bat | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libraries/cmark/nmake.bat b/libraries/cmark/nmake.bat
index 11d76c9..8d1c8b6 100644
--- a/libraries/cmark/nmake.bat
+++ b/libraries/cmark/nmake.bat
@@ -1 +1 @@
-@nmake.exe /nologo /f Makefile.nmake %*
+ 
